pkg/cluster/workloadidentityresources.go

```diff
@@ -0,0 +1,109 @@
+package cluster
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"fmt"
+
+	configv1 "github.com/openshift/api/config/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kruntime "k8s.io/apimachinery/pkg/runtime"
+)
+
+const azureFederatedTokenFileLocation = "/var/run/secrets/openshift/serviceaccount/token"
+
+func (m *manager) generateWorkloadIdentityResources() ([]kruntime.Object, error) {
+	if !m.doc.OpenShiftCluster.UsesWorkloadIdentity() {
+		return nil, fmt.Errorf("generateWorkloadIdentityResources called for a CSP cluster")
+	}
+
+	resources := []kruntime.Object{}
+	if platformWorkloadIdentitySecrets, err := m.generatePlatformWorkloadIdentitySecrets(); err != nil {
+		return nil, err
+	} else {
+		for _, secret := range platformWorkloadIdentitySecrets {
+			resources = append(resources, secret)
+		}
+	}
+
+	if cloudCredentialOperatorSecret, err := m.generateCloudCredentialOperatorSecret(); err != nil {
+		return nil, err
+	} else {
+		resources = append(resources, cloudCredentialOperatorSecret)
+	}
+
+	if authenticationConfig, err := m.generateAuthenticationConfig(); err != nil {
+		return nil, err
+	} else {
+		resources = append(resources, authenticationConfig)
+	}
+
+	return resources, nil
+}
+
+func (m *manager) generatePlatformWorkloadIdentitySecrets() ([]*corev1.Secret, error) {
+	if !m.doc.OpenShiftCluster.UsesWorkloadIdentity() {
+		return nil, fmt.Errorf("generatePlatformWorkloadIdentitySecrets called for a CSP cluster")
+	}
+
+	subscriptionId := m.subscriptionDoc.ID
+	tenantId := m.subscriptionDoc.Subscription.Properties.TenantID
+	region := m.doc.OpenShiftCluster.Location
+
+	roles := m.platformWorkloadIdentityRolesByVersion.GetPlatformWorkloadIdentityRolesByRoleName()
+
+	secrets := []*corev1.Secret{}
+	for _, identity := range m.doc.OpenShiftCluster.Properties.PlatformWorkloadIdentityProfile.PlatformWorkloadIdentities {
+		if role, ok := roles[identity.OperatorName]; ok {
+			secrets = append(secrets, &corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: role.SecretLocation.Namespace,
+					Name:      role.SecretLocation.Name,
+				},
+				Type: corev1.SecretTypeOpaque,
+				StringData: map[string]string{
+					"azure_client_id":            identity.ClientID,
+					"azure_subscription_id":      subscriptionId,
+					"azure_tenant_id":            tenantId,
+					"azure_region":               region,
+					"azure_federated_token_file": azureFederatedTokenFileLocation,
+				},
+			})
+		}
+	}
+
+	return secrets, nil
+}
+
+func (m *manager) generateCloudCredentialOperatorSecret() (*corev1.Secret, error) {
```

Comment:
What's the idea of using `UsesWorkloadIdentity` in generatePlatformWorkloadIdentitySecrets but not in generateAuthenticationConfig and generateCloudCredentialOperatorSecret?
Is it because the other 2 will always be called via generateWorkloadIdentityResources?

Reply:
This is the intent, yes. I forgot to remove the check within `generatePlatformWorkloadIdentitySecrets`, but otherwise, these functions are effectively private (though we can't really make them inaccessible to the entire rest of the `cluster` package) so they shouldn't be called outside of `generateWorkloadIdentityResources`.